apps/frontend/src/app/5_pages/MarketMakingPage/hooks/useGetUserInfo.ts

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { ethers } from 'ethers';
+
+import {
+  SupportedTokens,
+  getProtocolContract,
+  getTokenContract,
+} from '@sovryn/contracts';
+import { getProvider } from '@sovryn/ethers-provider';
+import { Decimal } from '@sovryn/utils';
+
+import { defaultChainId } from '../../../../config/chains';
+
+import { useAccount } from '../../../../hooks/useAccount';
+import { useGetProtocolContract } from '../../../../hooks/useGetContract';
+import { asyncCall } from '../../../../store/rxjs/provider-cache';
+import { decimalic } from '../../../../utils/math';
+import { UserInfo } from '../MarketMakingPage.types';
+import { AmmLiquidityPool } from '../utils/AmmLiquidityPool';
+
+export const useGetUserInfo = (pool: AmmLiquidityPool) => {
+  const { account } = useAccount();
+  const { assetA, assetB, poolTokenA, poolTokenB } = pool;
+  const [reward, setReward] = useState('0');
+  const [balanceA, setBalanceA] = useState(Decimal.ZERO);
+  const [loadingA, setLoadingA] = useState(false);
+  const [balanceB, setBalanceB] = useState(Decimal.ZERO);
+  const [loadingB, setLoadingB] = useState(false);
+  const liquidityMiningProxy = useGetProtocolContract('liquidityMiningProxy');
+  const babelfishAggregator = useGetProtocolContract('babelfishAggregator');
+
+  const getUserInfo = useCallback(
+    async (token: string) => {
+      return await asyncCall(
+        `liquidityMiningProxy/getUserInfo/${token}/${account}`,
+        () => liquidityMiningProxy?.getUserInfo(token, account),
+      ).then(res => ({
+        amount: res?.amount || '0',
+        reward: res?.accumulatedReward || '0',
+      }));
+    },
+    [account, liquidityMiningProxy],
+  );
+
+  const getContract = useCallback(async (token: string) => {
+    const { address, abi } = await getTokenContract(
+      token === SupportedTokens.rbtc ? SupportedTokens.wrbtc : token,
+      defaultChainId,
+    );
+    return new ethers.Contract(address, abi, getProvider(defaultChainId));
+  }, []);
+
+  const getBalance = useCallback(
+    async (info: UserInfo, totalSupply: Decimal, converterBalance: Decimal) => {
+      return decimalic(info.amount.toString())
+        .div(totalSupply)
+        .mul(converterBalance)
+        .toNumber()
+        .toFixed(0);
+    },
+    [],
+  );
```

Comment:
```suggestion
  const getBalance = useCallback(
    async (
      info: UserInfo,
      totalSupply: Decimal,
      converterBalance: Decimal,
    ): Promise<Decimal> =>
      Decimal.fromBigNumberString(info.amount)
        .div(totalSupply)
        .mul(converterBalance),
    [],
  );
```
We want to have `balanceA` and `balanceB` of type `Decimal`. Also, move this function above the hook since it has no dependencies and there's no need for it to be `async` because it just contains some basic calculations.